platform/spire/src/virt.py

```diff
@@ -339,7 +352,8 @@ def boot_with_io(self, phase, output_callback=None, text: bytes=None, delay=None
                 output = log_output
 
             if text:
-                stdin = delayed_yielder(delay, text)
+                assert delay is not None and chardelay is not None
```

Comment:
this is not what `assert` is for; see #303

Reply:
yeah, this is very much a "sanity-check my own code" assert